trajectory_follower_wrapper/test/test_trajectory_follower_wrapper.cpp

```diff
@@ -0,0 +1,209 @@
+    aw_cmd->longitudinal.acceleration = 1.0;
+    aw_cmd->longitudinal.speed = 2.0;
+    aw_cmd->lateral.steering_tire_angle = 0.5;
+
+    worker_node->ackermann_control_cb(move(aw_cmd));
+
+    auto res = worker_node->generate_command();
+    // resulting control command is empty because trajectory and twist member objects were not set
+    ASSERT_NEAR(res.cmd.linear_acceleration, 0.0, 0.0001);
+
+    std::unique_ptr<carma_planning_msgs::msg::TrajectoryPlan> traj_plan = std::make_unique<carma_planning_msgs::msg::TrajectoryPlan>();
+
+    carma_planning_msgs::msg::TrajectoryPlanPoint tpp, tpp2, tpp3;
+    tpp.x = 100;
+    tpp.y = 100;
+    tpp.target_time = rclcpp::Time(1.0*1e9);  // 14.14 m/s
+
+    tpp2.x = 110;
+    tpp2.y = 110;
+    tpp2.target_time = rclcpp::Time(2.0*1e9);  // 14.14 m/s
+
+    tpp3.x = 120;
+    tpp3.y = 120;
+    tpp3.target_time = rclcpp::Time(3.0*1e9);  // 14.14 m/s
+
+    carma_planning_msgs::msg::TrajectoryPlan plan;
+    traj_plan->initial_longitudinal_velocity = 14.14;
+    traj_plan->header.stamp = worker_node->now();
+    traj_plan->trajectory_points = { tpp, tpp2, tpp3 };
+
+    worker_node->current_trajectory_callback(move(traj_plan));
+
+    std::unique_ptr<geometry_msgs::msg::TwistStamped> twist_msg = std::make_unique<geometry_msgs::msg::TwistStamped>();
+    twist_msg->header.stamp = worker_node->now();
+    twist_msg->twist.linear.x = 4;
+    twist_msg->twist.linear.y = 0;
+    twist_msg->twist.linear.z = 0;
+    worker_node->current_twist_callback(move(twist_msg));
+
+    std::unique_ptr<geometry_msgs::msg::PoseStamped> pose_msg = std::make_unique<geometry_msgs::msg::PoseStamped>();
+
+    pose_msg->header.stamp = worker_node->now();
+    pose_msg->pose.position.x = 1;
+    pose_msg->pose.position.y = 2;
+    pose_msg->pose.position.z = 3;
+    worker_node->current_pose_callback(move(pose_msg));
+
+    worker_node->config_.incoming_cmd_time_threshold = 10.0;
+
+    // control successfully set since time threshold is increased and trajectory, pose and twist objects are set
+    auto res2 = worker_node->generate_command();
+
+    ASSERT_NEAR(res2.cmd.linear_acceleration, 1.0, 0.0001);
+    ASSERT_NEAR(res2.cmd.linear_velocity, 2.0, 0.0001);
+    ASSERT_NEAR(res2.cmd.steering_angle, 0.5, 0.0001);
+
+    worker_node->timer_callback();
+}
+
+TEST(Testtrajectory_follower_wrapper, TestThreshold){
+
+    rclcpp::NodeOptions options;
+    auto worker_node = std::make_shared<trajectory_follower_wrapper::TrajectoryFollowerWrapperNode>(options);
+
+    worker_node->configure(); //Call configure state transition
+    worker_node->activate();  //Call activate state transition to get not read for runtime
+
+    autoware_auto_msgs::msg::AckermannControlCommand aw_cmd;
+    aw_cmd.stamp = worker_node->now() - rclcpp::Duration::from_seconds(5);
+    aw_cmd.longitudinal.acceleration = 1.0;
+    aw_cmd.longitudinal.speed = 2.0;
+
+    bool res1 = worker_node->isControlCommandOld(aw_cmd);
```

Comment:
Prior to calling `isControlCommandOld()`, can we set `worker_node->config_.incoming_cmd_time_threshold` explicitly in this unit test to show the threshold value that is being tested here?

Reply:
updated